tn_api_v2_core.yml

```diff
@@ -0,0 +1,956 @@
+                total_price:
+                    type: number
+                taxes:
+                    type: number
+                tax_breakdown:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/DatasourceTaxBreakdownDetails'
+
+        DatasourceTaxBreakdownDetails:
+            type: object
+            description: Breakdown of taxes amount into their categories
+            properties:
+                category:
+                    type: string
+                    description: Tax category. Example = "CA"
+                amount:
+                    type: number
+                    description: Tax amount. Example = 5.295
+
+        GenerateSolutionsResponse:
+            title: Generate Solutions Response
+            type: object
+            properties:
+                trip_id:
+                    type: string
+                    description: A unique key used for identification and when booking a trip.
+                    example: "fb3d8eff432a329dc21fad8449c686fa93e5ce24"
+                itineraries:
+                    $ref: '#/components/schemas/DatasourceItineraryInfo'
+
+        DatasourceItineraryInfo:
+            type: array
+            description: Array of itineraries sorted by weight.
+            items:
+                properties:
+                    weight:
+                        type: float
+                        description: Weight of the segment (time-value considered).
+                        example: 239
+                    total_price:
+                        type: float
+                        description: Price of the itinerary (fare and tax included).
+                        example: 197.1
+                    transportation_time:
+                        type: integer
+                        description: Total transportation time for the itinerary
+                        example: 425
+                    structure:
+                        type: Array of ints and/or Array of tuple/or Array of tuple and int
+                        description: |
+                            "Array representing the structure of the trip in terms of one-ways and open-jaw segments. 
+                            For example: [(0, 1), (3, 4), 2] means there would be three bookings. One openjaw with key “01”, 
+                            one with key “34” and a one-way with key ‘2”."
+                            See Price Confirm Report segment_ids field for more detail.
+                        example: "[(0, 1), 2]"
+                    segment_metadata:
+                        type: array
+                        description: "Contains extra information about the segment that was sent by the OTA.
+                            This field is completely optional and we will not alter the data within this object.
+                            Whatever passed within this object will be returned as-is back to the OTA."
```

Comment:
For this response one, i think you should give a little explanation of like you did in the comments about why this is an array. And that it contains the metadata for for each segment composing the itinerary in the order they take place or something.

Im also wondering, because this segment_metadata is refering to the response. You may be able to remove the last two lines. As they seem to me to be refering to the request version of this variable.